folium/features.py

```diff
@@ -241,6 +241,14 @@ def __init__(self, data, style_function=None):
             else:  # This is a filename
                 self.embed = False
                 self.data = data
+        elif hasattr(data, 'to_crs'):
+            # We have a GeoPandas object
+            if hasattr(data, 'columns'):
+                # We have a GeoDataFrame
+                self.embed = True
+                self.data = json.loads(data.to_crs(epsg='4326').to_json())
```

Comment:
`GeoDataFrame`s have a `df.crs` that is empty when it it not defined.  We might take advantage of that to do some magic here.  The problem is that we might need `pyproj` for that...